### DISCUSSION TITLE
Work out the Amazon Region domain?

### DISCUSSION THREAD

**Discussion by Spookster** [alandtse/alexa_media_player#2780] [Q&A] [answered]
In the settings, what domain do I use? there isnt a .co.nz domain, and .com and .com.au dont work for me.

Is there another trick for set up? cheers :)

**Comment by danielbrunt57** [comment]
> there isn't a .co.nz domain

What do you mean by that?
When I browse to amazon.co.nz, it forwards/loads amazon.com.au.

Which domain do you log into when you browse to Amazon?
And are your devices registered in that domain?

![image](https://github.com/user-attachments/assets/ea37d04f-3420-4a39-9c05-1b839b1e4d76)

**Comment by danielbrunt57** [comment]
If your domain is `amazon.co.nz`, that should work as far as http2push is concerned but there's no locale key for that domain in alexapy.
If you log into amazon.com.au and you see your devices listed there then that should work. 

alexapy/const.py may need to be updated to include `".co.nz": "en-NZ"` in the `LOCALE_KEY`.

`alexapy/const.py`:
```

For more details about this api, please refer to the documentation at
https://gitlab.com/keatontaylor/alexapy
"""

EXCEPTION_TEMPLATE = "An exception of type {0} occurred. Arguments:\n{1!r}"

# Apollon77/alexa-remote - alexa-remote.js:
# const officialUserAgent = 'AppleWebKit PitanguiBridge/2.2.595606.0-[HARDWARE=iPhone14_7][SOFTWARE=17.4.1][DEVICE=iPhone]';

# CALL_VERSION = "2.2.556530.0"
CALL_VERSION = "2.2.595606.0"
APP_NAME = "Alexa Media Player"
# USER_AGENT = f"AmazonWebView/Amazon Alexa/{CALL_VERSION}/iOS/16.6/iPhone"
# USER_AGENT = f"AmazonWebView/Amazon Alexa/{CALL_VERSION}/iOS/17.4.1/iPhone"
# Trying...
SOFTWARE = "SOFTWARE=17.4.1"
USER_AGENT = f"AppleWebKit PitanguiBridge/{CALL_VERSION}-[HARDWARE=iPhone14_7][{SOFTWARE}][DEVICE=iPhone]"

LOCALE_KEY = {
    ".de": "de_DE",
    ".com.au": "en_AU",
    ".ca": "en_CA",
    ".co.uk": "en_GB",
    ".in": "en_IN",
    ".com": "en_US",
    ".es": "es_ES",
    ".mx": "es_MX",
    ".fr": "fr_FR",
    ".it": "it_IT",
    ".co.jp": "ja_JP",
    ".com.br": "pt_BR",
}
# https://developer.amazon.com/en-US/docs/alexa/alexa-voice-service/api-overview.html#endpoints
HTTP2_NA = "alexa.na.gateway.devices.a2z.com"
HTTP2_EU = "alexa.eu.gateway.devices.a2z.com"
HTTP2_FE = "alexa.fe.gateway.devices.a2z.com"
HTTP2_AUTHORITY = {
    ".com": HTTP2_NA,
    ".ca": HTTP2_NA,
    ".com.mx": HTTP2_NA,
    ".com.br": HTTP2_NA,
    ".co.jp": HTTP2_FE,
    ".com.au": HTTP2_FE,
    ".com.in": HTTP2_FE,
    ".co.nz": HTTP2_FE,
}
HTTP2_DEFAULT = HTTP2_EU
```

Modification:
```
LOCALE_KEY = {
    ".de": "de_DE",
    ".com.au": "en_AU",
    ".ca": "en_CA",
    ".co.uk": "en_GB",
    ".in": "en_IN",
    ".com": "en_US",
    ".es": "es_ES",
    ".mx": "es_MX",
    ".fr": "fr_FR",
    ".it": "it_IT",
    ".co.jp": "ja_JP",
    ".com.br": "pt_BR",
    ".co.nz": "en-NZ",
}
```

**Comment by Spookster** [comment]
The device shows in amazon.com, not .au

Still worth changing the .py file?

![image](https://github.com/user-attachments/assets/69cfddad-1b90-4f90-b72b-2a60f3d14f1b)

**Reply by danielbrunt57** [reply]
> The device shows in amazon.com, not .au

If you only have one device and it's registered in Amazon.com then the region domain for setting up AMP needs to be `amazon.com` but you said that did not work.  What happens when amazon.com is used?  What do your debug logs for `alexa` show?

**Comment by Spookster** [comment, marked as answer]
Oddly enough, it does now work with Amazon.com..... sigh.
Maybe just a delay in device registration process? idk..... but thanks for looking in!

**Comment by danielbrunt57** [comment]
Yeah, it probably was a delay in the left hand communicating with the right hand.
Happy to hear it's working now.

Happy New Year!
